### ISSUE TITLE
🐞[Execution failed for task ':workmanager:compileDebugKotlin'.]

### ISSUE THREAD

**Issue by loremed** [fluttercommunity/flutter_workmanager#379]
- [x] I have read the [README](https://github.com/vrtdev/flutter_workmanager/blob/master/README.md)
- [x] I have done the setup for [Android](https://github.com/vrtdev/flutter_workmanager/blob/master/ANDROID_SETUP.md)
- [ ] I have done the setup for [iOS](https://github.com/vrtdev/flutter_workmanager/blob/master/IOS_SETUP.md)
- [x] I have ran the sample app and it does not work there

**Version**

| Technology           | Version       |
| -------------------  | ------------- |
| Workmanager version  |       0.4.1        |
| Kotlin version        |         1.6.21     |

**Describe the error**

When launching both my app and the sample app provided, an error is returned:
```
e: /tmp/flutter_workmanager/android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt: (148, 21): Object is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result                                                                                                                                                                                
e: /tmp/flutter_workmanager/android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt: (153, 25): 'error' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':workmanager:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
Running Gradle task 'assembleDebug'...                              7.6s
Exception: Gradle task assembleDebug failed with exit code 1
```

This problem occurs with Kotlin versions 1.5.10, 1.5.21, 1.6.10, too.

The problem occurs as soon as workmanager is added as a dependency and not even used.
At this point, the issue seems to be that the module can't be compiled for some reason.

Flags like --info or --debug don't change the output.

**Output of `flutter doctor -v`**
```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.444, on Ubuntu 20.04.4 LTS 5.4.0-109-generic, locale en_US.UTF-8)
    • Flutter version 2.13.0-0.0.pre.444 at /opt/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 221235c7ed (4 weeks ago), 2022-04-08 03:01:08 -0700
    • Engine revision 80f807aa14
    • Dart version 2.17.0 (build 2.17.0-282.0.dev)
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /home/Lorenzo/Android/Sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 8.0.1-9 (tags/RELEASE_801/final)
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.1)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 66.0.1
    • Dart plugin version 211.7817
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.66.2)
    • VS Code at /usr/share/code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • moto g 7 plus (mobile) • ZY225CZV7N • android-arm64  • Android 10 (API 29)
    • Linux (desktop)        • linux      • linux-x64      • Ubuntu 20.04.4 LTS 5.4.0-109-generic
    • Chrome (web)           • chrome     • web-javascript • Google Chrome 101.0.4951.54

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found! 
```

**Comment by berkeugur** [comment]
`e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\workmanager-0.4.1\android\src\main\kotlin\be\tramckrijte\workmanager\BackgroundWorker.kt: (132, 21): Object 
is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result
e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\workmanager-0.4.1\android\src\main\kotlin\be\tramckrijte\workmanager\BackgroundWorker.kt: (137, 25): 'error' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':workmanager:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.`


i am getting the same error and i can't figure it out for days

**Comment by berkeugur** [comment]
flutter build apk --split-per-abi


e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\workmanager-0.4.1\android\src\main\kotlin\be\tramckrijte\workmanager\BackgroundWorker.kt: (132, 21): Object 
is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result
e: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\workmanager-0.4.1\android\src\main\kotlin\be\tramckrijte\workmanager\BackgroundWorker.kt: (137, 25): 'error' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':workmanager:compileReleaseKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BU�LD FAILED in 5m 22s
Running Gradle task 'assembleRelease'...                          324,0s
Gradle task assembleRelease failed with exit code 1

**Comment by Livinglist** [comment]
same issue on Flutter 3.0.0

**Comment by Livinglist** [comment]
temp fix: remove the `?` from first arg:

exact location is `.../flutter/.pub-cache/hosted/pub.dartlang.org/workmanager-0.4.1/android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt` line 137

```
                        override fun error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
                            Log.e(TAG, "errorCode: $errorCode, errorMessage: $errorMessage")
                            stopEngine(Result.failure())
                        }
```

**Comment by ened** [comment]
Please retry with V0.5.0

**Comment by berkeugur** [comment]
this issue has been fixed in v0.5.0. thank you to everyone who contributed.
